lib/config_files/grub2/default.rb

```diff
@@ -0,0 +1,176 @@
+require "config_files/base_model"
+require "config_files/augeas_parser"
+
+module ConfigFiles
+  module Grub2
+    # Represents grub configuration in /etc/default/grub
+    # Main features:
+    #
+    # - Do not overwrite files
+    # - When setting value first try to just change value if key already exists
+    # - When key is not set, then try to find commented out line with key and
+    #   replace it with real config
+    # - When even commented out code is not there, then append configuration
+    #   to the end of file
+    class Default < BaseModel
+      PARSER = AugeasParser.new("sysconfig.lns")
+      PATH = "/etc/default/grub"
+
+      def initialize(file_class: File)
+        super(PARSER, PATH, file_class: file_class)
+        self.data = AugeasTree.new
+      end
+
+      def os_prober
+        @os_prober ||= BooleanValue.new(
+          "GRUB_DISABLE_OS_PROBER",
+          self,
+          default:     true,
+          # grub key is disable, so use reverse logic
+          true_value:  "false",
+          false_value: "true"
+        )
+      end
+
+      def disable_recovery_entry
+        generic_set("GRUB_DISABLE_RECOVERY", "true")
+      end
+
+      def enable_recovery_entry(kernel_params)
+        generic_set("GRUB_DISABLE_RECOVERY", "false")
+        generic_set("GRUB_CMDLINE_LINUX_RECOVERY", kernel_params)
+      end
+
+      def timeout
+        data["GRUB_TIMEOUT"] || 10
```

Comment:
good question. It is actually grub2 default value. I do not know where else is such constant stored. probably only inside grub2 scripts or C files.